trino-parser/src/main/java/io/trino/sql/SqlFormatter.java

```diff
@@ -378,8 +379,14 @@ protected Void visitQuery(Query node, Integer indent)
 
             processRelation(node.getQueryBody(), indent);
             node.getOrderBy().ifPresent(orderBy -> process(orderBy, indent));
-            node.getOffset().ifPresent(offset -> process(offset, indent));
-            node.getLimit().ifPresent(limit -> process(limit, indent));
+            if (dialect.equals(BIGQUERY)) {
+                node.getLimit().ifPresent(limit -> process(limit, indent));
+                node.getOffset().ifPresent(offset -> process(offset, indent));
+            }
+            else {
+                node.getOffset().ifPresent(offset -> process(offset, indent));
+                node.getLimit().ifPresent(limit -> process(limit, indent));
+            }
```

Comment:
Does the order matters ? Looks like both if-else blocks are almost doing the same thing.

Reply:
Yes, it's an ordering problem. BigQuery need to place `OFFSET` behind `LIMIT`.